### DISCUSSION TITLE
Context tracking

### DISCUSSION THREAD

**Discussion by ChristopheQuintard** [activej/activej#293] [Ideas]
Hi,
I'm implementing microservices using ActiveJ, but I run into a problem. For each incoming request, I request a JWT to authentify and authorize the user that makes the request. The processing of the request involves making requests to other services. In order to achieve in-depth security, I need to forward the JWT to the invoked services. But I don't know how to propagate the JWT from method to method, through Promise and ChannelSupplier. I thought I could use a HTTP scope, but it is not implemented, it is just mentioned in an example. 
It would be great to have something similar to NodeJS's context tracking : https://nodejs.org/api/async_context.html#asynchronous-context-tracking
It is something very delicate to implement, but it would really help.
Thanks, Christophe

**Comment by ChristopheQuintard** [comment]
Oups, it seems the same issue has been raised twice but phrased differently.
OpenTelemtry integration and promise contexts #264
ActiveJ http and MDC/Thread local #265
Using HttpMessage is not a solution for me.

**Comment by eduard-vasinskyi** [comment]
Hi, @ChristopheQuintard

We still have no plans to add context to promises.

If you really need to pass context around, you need to implement context handling yourself. One solution could be to wrap your data into a class containing the context.
